compiler/rustc_trait_selection/src/traits/const_evaluatable.rs

```diff
@@ -609,9 +609,27 @@ where
 /// Tries to unify two abstract constants using structural equality.
 pub(super) fn try_unify<'tcx>(
     tcx: TyCtxt<'tcx>,
-    a: AbstractConst<'tcx>,
-    b: AbstractConst<'tcx>,
+    mut a: AbstractConst<'tcx>,
+    mut b: AbstractConst<'tcx>,
 ) -> bool {
+    while let Node::Leaf(a_ct) = a.root() {
```

Comment:
Leave a comment here that this is essentially performing generic substitution, but on the fly.

Reply:
I didn't write a comment saying *exactly* that but there's now a comment, hope that's fine too ^^